src/test/codewhisperer/util/crossFileContextUtil.test.ts

```diff
@@ -0,0 +1,168 @@
+            mockEditor = createMockTextEditor('content', 'fileName', 'cpp')
+            const actual = await crossFile.fetchSupplementalContextForSrc(mockEditor, fakeCancellationToken)
+            assert.strictEqual(actual, undefined)
+        })
+
+        it('ruby', async function () {
+            mockEditor = createMockTextEditor('content', 'fileName', 'ruby')
+
+            const actual = await crossFile.fetchSupplementalContextForSrc(mockEditor, fakeCancellationToken)
+
+            assert.strictEqual(actual, undefined)
+        })
+    })
+
+    describe('partial support - control group', function () {
+        before(async function () {
+            this.timeout(60000)
+        })
+
+        beforeEach(async function () {
+            tempFolder = (await createTestWorkspaceFolder()).uri.fsPath
+        })
+
+        afterEach(async function () {
+            sinon.restore()
+            userGroupSettings.reset()
+            await closeAllEditors()
+        })
+
+        it('should be empty if userGroup is control', async function () {
+            if (!shouldRunTheTest) {
+                this.skip()
+            }
+
+            sinon.stub(userGroupSettings, 'userGroup').returns(UserGroup.Control)
+
+            // TODO: flaky, is there other approach to achive this? need this probably because it's an async test
+            while (userGroupSettings.userGroup !== UserGroup.Control) {}
+
+            const editor = await openATextEditorWithText('content-1', 'file-1.js', tempFolder)
+            await openATextEditorWithText('content-2', 'file-2.js', tempFolder)
+            await openATextEditorWithText('content-3', 'file-3.js', tempFolder)
+            await openATextEditorWithText('content-4', 'file-4.js', tempFolder)
+
+            await assertTabCount(4)
+
+            const actual = await crossFile.fetchSupplementalContextForSrc(editor, fakeCancellationToken)
+
+            assert.ok(actual !== undefined && actual.length === 0)
+        })
+    })
+
+    describe('partial support - crossfile group', function () {
+        before(async function () {
+            this.timeout(60000)
+        })
+
+        beforeEach(async function () {
+            tempFolder = (await createTestWorkspaceFolder()).uri.fsPath
+        })
+
+        afterEach(async function () {
+            sinon.restore()
+            userGroupSettings.reset()
+            await closeAllEditors()
+        })
+
+        it('should be non empty if userGroup is crossfile', async function () {
+            if (!shouldRunTheTest) {
+                this.skip()
+            }
+
+            sinon.stub(userGroupSettings, 'userGroup').returns(UserGroup.CrossFile)
+
+            while (userGroupSettings.userGroup !== UserGroup.CrossFile) {}
+
+            // TODO: flaky, is there other approach to achive this? need this probably because it's an async test
+            console.log(`in crossfile group test, ${userGroupSettings.userGroup}`)
```

Comment:
@justinmk3 @JadenSimon, naive q, I need this waiting probably because it's an async test, is there better approach to deal with it? thanks!